iothub/service/src/Amqp/AmqpClientHelper.cs

```diff
@@ -114,69 +109,52 @@ internal static Exception ToIotHubClientContract(Error error)
 
             if (error.Condition.Equals(IotHubAmqpErrorCode.TimeoutError))
             {
-                retException = new IotHubServiceException(message, HttpStatusCode.RequestTimeout, IotHubServiceErrorCode.Unknown);
+                retException = new(message, HttpStatusCode.RequestTimeout, IotHubServiceErrorCode.Unknown);
             }
             else if (error.Condition.Equals(AmqpErrorCode.NotFound))
             {
-                retException = new IotHubServiceException(message, HttpStatusCode.NotFound, IotHubServiceErrorCode.DeviceNotFound);
-            }
-            else if (error.Condition.Equals(AmqpErrorCode.NotImplemented))
-            {
-                retException = new NotSupportedException(message);
-            }
-            else if (error.Condition.Equals(AmqpErrorCode.NotAllowed))
-            {
-                retException = new InvalidOperationException(message);
```

Comment:
similarly here, these will be addressed in the "else" block below and returned as `IotHubServiceException`.